File: java/warmup1/posNeg.java
package java.warmup1;

public class posNeg {
  public boolean posNeg(int a, int b, boolean negative) {

    if (negative) {
      if (a < 0 && b < 0)
        return true;
      else
        return false;

    } else {
      if (a < 0 && b > 0)
        return true;
      else if (b < 0 && a > 0)
        return true;
      else
        return false;

    }

  }

}

File: python/warmup1/pos_neg.py
def pos_neg(a, b, negative):

    return (negative and (a < 0 and b < 0)) or ((not negative) and
                                                ((a < 0 and b > 0) or
                                                 (a > 0 and b < 0)))

    # alternatively:

    #if(negative):
    #return a < 0 and b < 0
    #else:
    #return (a < 0 and b > 0) or (a > 0 and b < 0)